### ISSUE TITLE
Some question about the paper

### ISSUE THREAD

**Issue by wy9933** [SCUT-BIP-Lab/PointDC#1]
Hello, 

Thanks for your good work!

As your code hasn't been uploaded yet, I've encountered a few questions about the paper that I couldn't address through the code. That's why I've raised this issue.

1. The paper mentions in section 3.4 that `the output feature dimension of fθ(·) is D`, but in Eq.11, the cross entropy function directly computes the loss between the D-dimensional features and the C-dimensional classification labels.Is there a classification layer applied before this computation?

2. In `CVPR 2023 GrowSP: Unsupervised Semantic Segmentation of 3D Point Clouds`, the results of `PiCIE` is greater than your experiment, is this due to parameter adjustments?

Thanks once again for your work; looking forward to your response.

**Comment by HalvesChen** [comment]
Thanks for your interest in our ***@***.***
            
        
     
        
  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            Yi ***@***.***>
            
        
      
      
        
         Date 
        
        
        12/27/2023 21:24
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              [SCUT-BIP-Lab/PointDC] Some question about the paper (Issue #1)
        
      
    
  

Hello,
Thanks for your good work!
As your code hasn't been uploaded yet, I've encountered a few questions about the paper that I couldn't address through the code. That's why I've raised this issue.


The paper mentions in section 3.4 that the output feature dimension of fθ(·) is D, but in Eq.11, the cross entropy function directly computes the loss between the D-dimensional features and the C-dimensional classification labels.Is there a classification layer applied before this computation?


In CVPR 2023 GrowSP: Unsupervised Semantic Segmentation of 3D Point Clouds, the results of PiCIE is greater than your experiment, is this due to parameter adjustments?


Thanks once again for your work; looking forward to your response.

**Comment by HalvesChen** [comment]
Hello, Thanks for your interest in our work! 
Regarding your first question: We use the $\mu$  obtained from clustering in Equation 8 as the weight for the classification head to calculate the probabilities of each supervoxel belonging to a certain centroid. This is an oversight in our paper, and we will update the paper on arXiv to address this.

For your second question: I believe the main differences lie in the experimental settings. We have reviewed the code for growsp, and there are some differences in both model design (we use Unet, inspired by PointGroup, while they use an FPN structure) and implementation (we use an older version of spconv, whereas growsp uses minkowski). Additionally, PiCIE is based on a contrastive learning framework, which is sensitive to data augmentation. There may be some differences in how to handle data augmentation.

We have also tested our method using growsp's model, and the final results were slightly higher than pointdc. We plan to release the subsequent code gradually. Thank you again for your attention.
